lib/ruleHelper.js

```diff
@@ -50,54 +50,59 @@ RuleHelper.prototype = {
         */
 
 
-        /*  surely, someone could have an evil literal in there, but that"s malice
-        we can just check for unsafe coding practice, not outright malice
-        example literal "<script>eval(location.hash.slice(1)</script>"
-        (it"s the task of the tagger-function to be the gateway here.)
-        */
-        if (expression.type === "Literal") {
-            // we just assign a literal (e.g. a string, a number, a bool)
+        switch(expression.type) {
+        case"Literal":
+            /*  surely, someone could have an evil literal in there, but that"s malice
+            we can just check for unsafe coding practice, not outright malice
+            example literal "<script>eval(location.hash.slice(1)</script>"
+            (it"s the task of the tagger-function to be the gateway here.)
+            */
             allowed = true;
-        } else if (expression.type === "TemplateElement") {
+            break;
+        case "TemplateElement":
             // Raw text from a template
             allowed = true;
-        } else if (expression.type === "TemplateLiteral") {
-            allowed = true;
+            break;
+        case "TemplateLiteral":
 
-            // check for ${..} expressions
-            for (let e = 0; e < expression.expressions.length; e++) {
-                const templateExpression = expression.expressions[e];
-                if (!this.allowedExpression(templateExpression, escapeObject, details)) {
-                    allowed = false;
-                    break;
-                }
-            }
-        } else if (expression.type === "TaggedTemplateExpression") {
+            // check only the ${..} expressions
+            allowed = this.allowedExpression(expression.expressions, escapeObject, details);
+            break;
+        case "TaggedTemplateExpression":
```

Comment:
I didn't expect this part to be part of the diff, this is definitely coming from #181, I guess that it may be rendered here because the recent tip has been merged back into the `fix-178-variabletracing-config` branch.

Locally I tried to get rid of it using `git rebase -i master`, which initially proposed me this unexpected list:

```
pick 816d9ee Introduce new configuration 'variableTracing', default to disabled. Adding tests for both states (fix #178)
pick 8f76655 Transform 'variableTracing' option into bool
pick ba64718 mention variableTracing in docs
pick f25a642 Remove spurious test case
pick 27faa93 Introduce new configuration 'variableTracing', default to disabled. Adding tests for both states (fix #178)
pick 4cbfb33 Transform 'variableTracing' option into bool
pick a6c4964 mention variableTracing in docs
pick b23a30f Remove spurious test case
```

accepting the proposed list as is makes the rebase to fail, which was surprising, it seems to be because he isn't able to figure out that the second groups of changes should be replacing the first group with the same commit names (but different commit hashes, as expected).

Dropping the first half of that list, and leave as `pick` the commits between the one with hash 27faa93 and the one with hash b23a30f, makes the rebase to complete successfully and to include only the expected changes in the resulting interdiff.

I'm not sure if the current status of the branch may confuse github merge button as well, but may be reasonable to fix that locally upfront as described above and update this branch on github before merging it.

Reply:
thanks for the debug hints. This was a weird mess. Should be good now :relieved: